### PR TITLE
Support reading and storing ReplayGain metadata

### PR DESCRIPTION
**Pull request by lstrsrt**
We don't do anything with these fields yet and they don't show up in the property viewer, but at least they're in the database now. This is step 1 towards resolving #129. Applying these fields in the playback engine should be pretty simple. I'll see what I can do tomorrow.

### CODE REVIEW

**Comment by lstrsrt** [comment]
Oops, will fix the build in a second.

I also investigated calculating ReplayGain values. The math looks difficult and highly error-prone so I think pulling in libebur128 might be a better idea than writing the code. It could also be an optional dependency, so when it's not available we just don't give the user the option to calculate RG values.

**Comment by diizzyy** [comment]
This can be done by using FFmpeg ( https://github.com/cyanreg/cyanrip/commit/5285205fda8812a07d3f056a336517c094181e39 ) which would be nice to avoid external dependencies.

**Comment by lstrsrt** [comment]
I see, thanks. Somehow didn't see that comment about ffmpeg.

**Review by ludouzi — CHANGES_REQUESTED** [review]

**Comment by ludouzi** [comment]
Thanks for making a start on this.

The fields were actually already stored as strings in the database in the extra tags blob, though it makes sense to store them in their own columns.

I agree with @diizzyy that we should avoid bringing in another dependency if possible. Or make it a plugin if not.

File: src/core/engine/taglibparser.cpp [hunk 1]
```diff
@@ -576,6 +576,20 @@ void readAudioProperties(const TagLib::File& file, Fooyin::Track& track)
...
 void readGeneralProperties(const TagLib::PropertyMap& props, Fooyin::Track& track)
 {
+    const auto gainStringToFloat = [](const TagLib::String& gainString) -> float {
+        const auto& string = convertString(gainString);
+        if(string.size() <= 1) {
+            return 0.0;
+        }
+
+        if(string.endsWith(QStringLiteral("dB"), Qt::CaseInsensitive)) {
+            return string.chopped(2).toDouble();
```
[the remark this answers is not in the document]
Comment (lstrsrt) [reply src/core/engine/taglibparser.cpp:586]: Oh yeah, I was using a double for these values earlier and forgot to convert all the function calls, thanks. There's also an "if" on line 670 that should be an "else if" which I'll change as well.

**Comment by lstrsrt** [comment]
If ffmpeg can be used for this there's definitely no need for a new dependency.

Does this change also warrant a database reload message?

**Comment by ludouzi** [comment]
> Does this change also warrant a database reload message?

It will yeah so the replaygain data in the extra tags blob is removed.

**Comment by lstrsrt** [comment]
Done 👍

**Review by ludouzi — APPROVED** [review]

**Pull request merged**